service/storage_service.hh

```diff
@@ -954,6 +956,7 @@ public:
     // if request completes with an error
     future<sstring> wait_for_topology_request_completion(utils::UUID id, bool require_entry = true);
     future<> wait_for_topology_not_busy();
+    std::vector<canonical_mutation> build_mutation_from_join_params(const join_node_request_params& params, api::timestamp_type write_timestamp);
```

Comment:
I don't understand why you made this function public. I don't see any new calls to it.